test/unit/spec/narrativeLoginSpec.js

```diff
@@ -1,19 +1,113 @@
-define (
-	[
-		'jquery',
-		'narrativeLogin'
-	], function(
-		$,
-		Login
-	) {
-    describe('Test the kbaseNarrative module', function() {
-        it('Should do things', function() {
-            expect(Login.init).not.toBe(null);
+/* global describe, it, expect, jasmine, beforeEach, afterEach */
+define ([
+    'jquery',
+    'narrativeLogin',
+    'narrativeConfig'
+], (
+    $,
+    Login,
+    Config
+) => {
+    'use strict';
+
+    const cookieKeys = ['kbase_session'],
+        FAKE_TOKEN = 'some_fake_token';
+
+    function setToken(token) {
+        cookieKeys.forEach((key) => {
+            document.cookie = `${key}=${token}`;
+        });
+    }
+
+    function clearToken() {
+        cookieKeys.forEach((key) => {
+            document.cookie = `${key}=`;
+        });
+    }
+
+    /**
+     *
+     * @param {string} request the path request
+     * @param {object} responseObj the response to send
+     * @param {int} status the status code to return
+     */
+    function mockAuthRequest(request, responseObj, status) {
+        let reqUrl = `${Config.url('auth')}/api/V2/${request}`;
+        jasmine.Ajax.stubRequest(reqUrl)
+            .andReturn({
+                status: status,
+                contentType: 'application/json',
+                responseText: JSON.stringify(responseObj)
+            });
+    }
+
+    describe('Test the kbaseNarrative module', () => {
+        beforeEach(() => {
+            setToken(FAKE_TOKEN);
+            jasmine.Ajax.install();
+        });
+
+        afterEach(() => {
+            jasmine.Ajax.uninstall();
+            clearToken();
+        });
+
+        it('Should instantiate and have expected functions', () => {
+            expect(Login).toBeDefined();
+            ['init', 'sessionInfo', 'getAuthToken', 'clearTokenCheckTimers'].forEach(item => {
+                expect(Login[item]).toBeDefined();
+            });
+        });
+
+        it('Should instantiate on a DOM node in the happy path', () => {
+            const $node = $('<div>'),
+                fakeUser = 'some_user',
```

Comment:
This blob of boilerplate should get added to the mocks module when that gets ported over from the `truss` branch.